hibernate-reactive-core/src/main/java/org/hibernate/reactive/id/impl/ReactiveCompositeNestedGeneratedValueGenerator.java

```diff
@@ -0,0 +1,108 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.id.impl;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.generator.BeforeExecutionGenerator;
+import org.hibernate.generator.Generator;
+import org.hibernate.generator.GeneratorCreationContext;
+import org.hibernate.id.CompositeNestedGeneratedValueGenerator;
+import org.hibernate.id.IdentifierGenerationException;
+import org.hibernate.mapping.Component;
+import org.hibernate.metamodel.spi.RuntimeModelCreationContext;
+import org.hibernate.reactive.id.ReactiveIdentifierGenerator;
+import org.hibernate.reactive.session.ReactiveConnectionSupplier;
+import org.hibernate.reactive.tuple.entity.ReactiveEntityMetamodel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import static org.hibernate.generator.EventType.INSERT;
+import static org.hibernate.reactive.util.impl.CompletionStages.completedFuture;
+import static org.hibernate.reactive.util.impl.CompletionStages.loop;
+
+public class ReactiveCompositeNestedGeneratedValueGenerator extends CompositeNestedGeneratedValueGenerator implements
+		ReactiveIdentifierGenerator<Object> {
+
+	public ReactiveCompositeNestedGeneratedValueGenerator(
+			CompositeNestedGeneratedValueGenerator generator,
+			GeneratorCreationContext creationContext,
+			RuntimeModelCreationContext runtimeModelCreationContext) {
+		super(
+				generator.getGenerationContextLocator(),
+				generator.getCompositeType(),
+				extracted( generator, creationContext, runtimeModelCreationContext )
```

Comment:
COuld you change the name of this method (`extracted`), please? `reactivePlans` for example